lib/active_merchant/billing/gateways/flo2cash_rest.rb

```diff
@@ -0,0 +1,595 @@
+      #   + Suspended can only be applied to active plans
+      #
+      def update_direct_debit_plan(plan_id, status, options)
+        params = { plan_id: plan_id, status: status }
+        commit(:put, :update_direct_debit_plan, params, options)
+      end
+
+      private
+
+      def commit(verb, action, params, options={})
+        begin
+          raw_response = ssl_request(verb, url(action, params), post_data(verb, params), headers(options))
+
+          response = parse(raw_response)
+          succeeded = success_from(response)
+        rescue ActiveMerchant::ResponseError => e
+          raise if e.response.code !~ CLIENT_ERROR_REGEX
+
+          response = parse(e.response.body)
+          succeeded = false
+        end
+
+        Response.new(
+          succeeded,
+          message_from(succeeded, response),
+          response,
+          authorization: authorization_from(action, response),
+          error_code: error_code_from(succeeded, response),
+          test: test?
+        )
+      end
+
+      def headers(options={})
+        {
+          'Content-Type' => 'application/json',
+          'Authorization' => "Basic #{options[:api_key]}"
+        }
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def success_from(response)
+        !response.key?('status') || SUCCESS_STATUSES.include?(response['status'])
+      end
+
+      def message_from(succeeded, response)
+        succeeded ? 'Succeeded' : error_code_from(false, response)
+      end
+
+      # Returns a card token
+      #
+      # ==== Options
+      #
+      # * <tt>payment_method</tt> -- The instrument to charge.
+      # * <tt>options</tt> -- A hash of parameters.
+      #
+      def retrieve_token(payment_method, options)
+        if payment_method.is_a?(String)
+          payment_method
+        else
+          store(payment_method, options).authorization
+        end
+      end
+
+      def add_customer_resume(post, options)
+        post[:initiatedBy] = options[:initiatedBy]
+        post[:receiptRecipient] = options[:email]
+      end
+
+      def add_customer_data(post, options)
+        post[:payer] = {
+          companyName: options[:company_name],
+          title: options[:title],
+          firstNames: options[:first_name],
+          lastName: options[:last_name],
+          dateOfBirth: options[:dob],
+          telephoneHome: options[:home_phone],
+          telephoneWork: options[:work_phone],
+          telephoneMobile: options[:mob_phone],
+          fax: options[:fax],
+          email: options[:email]
+        }
+
+        add_references(post, options)
+      end
+
+      def add_device_info(post, options)
+        post[:device] = {
+          id: options[:device_id],
+          description: options[:device_description]
+        }
+      end
+
+      def add_address(post, options)
+        post[:payer].merge({
+          address1: options.try(:[], :address1),
```

Comment:
Curious as to why you have used the `.try()` method on every instance here but not in other cases, like in the `add_customer_data` method for example?

Reply:
it's there because `add_address` is not being called with the raw `options` hash. It's being called with `options[:address]` hash and if it's empty by some reason, it's going to fail to try to access here.